orchestration/orchestration_taskdef.go

```diff
@@ -233,21 +249,81 @@ func (o *Orchestrator) DeleteTaskDef(ctx context.Context, input *TaskDefDeleteIn
 		}(taskDefinitionRevisions[1:])
 	}
 
-	deletedCluster, err := o.deleteCluster(ctx, &input.Cluster)
-	if err != nil {
-		return nil, fmt.Errorf("failed to delete cluster: %s", err)
-	}
+	// stop the running tasks and cleanup in the background
+	go func() {
+
+		// create a new context for the cleanup
+		cleanupCtx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		if l := len(runningTasks); l > 0 {
+			taskIds := make([]string, len(runningTasks))
+			reason := fmt.Sprintf("Deleting task definition %s", input.TaskDefinition)
+
+			// set desired status for all tasks to STOPPED
+			for i, t := range runningTasks {
+				splitTask := strings.SplitN(t, "/", 2)
+				taskIds[i] = splitTask[1]
+
+				if _, err := o.ECS.StopTask(cleanupCtx, &ecs.StopTaskInput{
+					Cluster: aws.String(input.Cluster),
+					Reason:  aws.String(reason),
+					Task:    aws.String(splitTask[1]),
+				}); err != nil {
+					log.Errorf("failed calling StopTask for %s", t)
+					return
+				}
+			}
+
+			// wait for tasks to become STOPPED
+			if err := retry(10, 10*time.Second, func() error {
```

Comment:
Code smell. What's the upper limit on aws for attempts to stop a task?

Reply:
(maximum time, not retries, sorry)

Reply:
I'm not sure I understand what you're getting at, this retries the function 10 times with a sliding window.

Reply:
Ahh nevermind, ignore that comment.